### ISSUE TITLE
node-http2 v4 via fork ?

### ISSUE THREAD

**Issue by hthetiot** [molnarg/node-http2#250]
Hello, 

We started a fork that include cumulative fixes.

See dedicated issue here:  https://github.com/kaazing/node-http2/issues/3

Here is a quick overview:

> This fork of node-http2 module start at version 4.0.0 in case previous repository pick-up work on version 3.x.x.

Changes:

* Rename npm package from 'http2' to 'node-http2' to avoid conflict.
* Fork from abandoned https://github.com/molnarg/node-http2 to https://github.com/kaazing/node-http2 and change version to '4.0.0' to avoid confusion.
* Bugfixes from pull requests
    - Clean up ended streams to free leaked memory (https://github.com/molnarg/node-http2/pull/200)
    - Memory leak fix (by argon) and another 'upstream is not defined' check (https://github.com/molnarg/node-http2/pull/222)
    - Allow frames to send after the arrival of the RST_STREAM. (https://github.com/molnarg/node-http2/pull/210)
    - GOAWAY deserialization fix (https://github.com/molnarg/node-http2/pull/235)
    - Update Mocha link (https://github.com/molnarg/node-http2/pull/229)
    - Remove invalid assert (https://github.com/molnarg/node-http2/pull/236)
    - Typo: finshed (https://github.com/molnarg/node-http2/pull/199)
    - JSHINT all source (https://github.com/dpwspoon/node-http2/pull/4)
    - Changed API to be fully pluggable for any transport or server (https://github.com/dpwspoon/node-http2/pull/2)

**Comment by akc42** [comment]
are you aware that node 8.4.0 now has experimental http2 support behind a --expose-http2 flag.  It is much faster as it relies on nghttp2 under the hood which is native c (or c++?) code.  I used to use this module, but have now switched over to the node one.  There is a core layer that gives access to the underlying duplex streams and a compat layer that makes it as close to http 1 as possible.

Because it is behind an experimental flag the api is still changing, but it would be good for as many people as possible to use it and comment on whether the api works.

**Comment by dpwspoon** [comment]
Yes, we are aware.  We use this library in production in environments where this support is not yet available AND on client platforms that ARE NOT node based.

As such, we will continue to support node-http2 for some time until many various client platforms offer full http2 support.

**Comment by hthetiot** [comment]
Example of migration:
- https://github.com/kaazing/http2.js/blob/develop/HISTORY.md#401-2017-10-01

**Comment by hthetiot** [comment]
4.0.3 (2018-07-18)
Update http.js to add support on request for retry-after header on 503|429|302 status code.

[About 503] The server returns an HTTP error 503 when more users than are allowed by the server's request queue limit have sent requests to a single server or when the actions of the client have triggered throttling.

Retry-After If a Retry-After header ([RFC2616]) is present in the response, the client SHOULD<6> retry the request after waiting the number of seconds indicated by the Retry-After header. Any such value represents an estimate of when the server is expected to be able to process the request.

- https://www.npmjs.com/package/http2.js
- https://github.com/kaazing/http2.js
